test/hotspot/jtreg/vmTestbase/nsk/jdb/kill/kill001/kill001.java

```diff
@@ -98,25 +99,38 @@ protected void runCases() {
         Vector v;
         String found;
         String[] threads;
+        boolean vthreadMode = "Virtual".equals(System.getProperty("main.wrapper"));
 
         jdb.setBreakpointInMethod(LAST_BREAK);
         reply = jdb.receiveReplyFor(JdbCommand.cont);
 
-        threads = jdb.getThreadIds(DEBUGGEE_THREAD);
+        // At this point we are at the breakpoint triggered by the breakHere() call done
+        // after creating all the threads. Get the list of debuggee threads.
+        threads = jdb.getThreadIdsByName(MYTHREAD);
 
         if (threads.length != numThreads) {
             log.complain("jdb should report " + numThreads + " instance of " + DEBUGGEE_THREAD);
             log.complain("Found: " + threads.length);
             success = false;
         }
 
+        // Kill each debuggee thread. This will cause each thread to stop in the debugger,
+        // indicating that an exception was thrown.
         for (int i = 0; i < threads.length; i++) {
+            // kill (ThreadReference.stop) is not supproted for vthreads, so we expect an error.
+            String msg = (vthreadMode ? "Illegal thread state" : "killed");
             reply = jdb.receiveReplyForWithMessageWait(JdbCommand.kill + threads[i] + " " +
                                                        DEBUGGEE_EXCEPTIONS + "[" + i + "]",
-                                                       "killed");
+                                                       msg);
         }
 
-        for (int i = 0; i <= numThreads; i++) {
+        // Continue the main thread, which is still at the breakpoint. This will resume
+        // all the debuggee threads, allowing the kill to take place.
+        reply = jdb.receiveReplyFor(JdbCommand.cont);
+
+        // Continue each of the threads that received the "kill" exception. Not needed
+        // for the vthread case since they are not actually killed.
+        if (!vthreadMode) for (int i = 0; i < numThreads; i++) {
```

Comment:
if and for at the same line don't look good

Reply:
ok